neutron_lbaas_dashboard/api/rest/esd.py

```diff
@@ -100,10 +108,70 @@ def dissociate_esd(request, l7policy_id):
 
 
 @urls.register
-class ESDs(generic.View):
+class RepoESDs(generic.View):
     """API for associating, retrieving ESD associated with a listener.
 
     """
+    url_regex = r'esd/$'
+
+    @rest_utils.ajax()
+    def get(self, request):
+        """List esd containers.
+
+        The listing result is an object with property "items".
+        """
+
+        esds = {}
+        esddir = '/etc/openstack-dashboard/esds'
+        try:
+            files = os.listdir(esddir)
+        except Exception as e:
+            errmsg = _("Failed to list directory '%(esddir)s': %(err)s")\
+                % {'esddir': esddir, 'err': e}
+            LOG.error(errmsg)
+            esds['_FAULT_FOLDER'] = {
+                'status': 'Folder Error',
+                'description': errmsg
+            }
+            return {'items': esds}
+
+        for file in files:
+            path = os.path.join(esddir, file)
+            try:
+                with open(path) as fr:
+                    esdjson = json.load(fr)
+                    for k, v in esdjson.items():
+                        if k in esds:
+                            warnmsg = _("Duplicate esd: %s, overriden.") % k
```

Comment:
Duplicate --> Duplicated ?

Reply:
Just noticed.. However, seen from 'http://dict.cn/Duplicate' and 'http://dict.cn/Duplicated', seem duplicate is better.